resolver-dns/src/main/java/io/netty/resolver/dns/DnsQueryContext.java

```diff
@@ -278,7 +278,7 @@ private void onQueryWriteCompletion(final long queryTimeoutMillis,
         }
 
         // Schedule a query timeout task if necessary.
-        if (queryTimeoutMillis > 0) {
+        if (queryTimeoutMillis > 0 && queryTimeoutMillis < Long.MAX_VALUE) {
```

Comment:
Why we need this change ? It is not clear to me after reading the description

Reply:
`Invalid arguement` is caused by a scheduled task that has `queryTimeoutMillis = Long.MAX_VALUE`.
I agree that this is an additional change but I don't see we need to start a scheduler for `Long.MAX_VALUE` which I and other users could use to disable the timeout.

Reply:
why not just use `0` if you want  to disable it ?

Reply:
It is possible. The Javadoc doesn't explain the input value, so I just randomly chose between 0 and Long.MAX_VALUE.
Anyway, this part is not directly related to the bug. Let me revert this.

Reply:
I realized that I wasn't able to set 0 as the query timeout so `Long.MAX_VALUE` was used as a workaround.
https://github.com/netty/netty/blob/830166292be28e64ec6b43170cff40a31a442ad4/resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java#L429-L431

Reply:
Created a separate PR https://github.com/netty/netty/pull/13505 to handle that issue.